concept/thing/impl/ThingImpl.java

```diff
@@ -122,7 +122,7 @@ public void setHas(Attribute attribute, boolean isInferred) {
                 throw exception(TypeDBException.of(THING_KEY_OVER, attribute.getType().getLabel(), getType().getLabel()));
             } else if (attribute.getOwners(getType()).first().isPresent()) {
                 throw exception(TypeDBException.of(THING_KEY_TAKEN, ((AttributeImpl<?>) attribute).getValue(),
-                        attribute.getType().getLabel(), getType().getLabel()));
+                        attribute.getType().getLabel(), attribute.getOwners(getType()).first().get().getType().getLabel()));
```

Comment:
Make sure the message reports the base type that owns the key, i.e. "owned by another 'base'", not "owned by another 'derived'". The latter can be misleading.